frontend/src/domain/WeclomeScreen/Form/Form.tsx

```diff
@@ -40,68 +75,114 @@ const useStyles = makeStyles((theme) => ({
-  label: string;
-  inputProps?: React.ComponentProps<typeof Input>;
-};
-
-export type FormProps = {
-  submitTitle: string;
-  footer: FooterProps;
-  formElements: Array<InputProps>;
+const validator = (elements: Array<InputProps>) => (
+  values: Record<string, any>
+) => {
+  const errors: Record<string, string> = {};
+  elements.forEach(({ label }) => {
+    if (!values[label]) {
+      errors[label] = "Required";
+    } else if (
+      label === "email" &&
+      !/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(
+        values[label].toLowerCase()
+      )
+    ) {
+      errors[label] = "Must be a mail";
+    } else if (
+      label === "repeat password" &&
+      values[label] !== values["password"]
+    ) {
+      errors[label] = "Must match";
+    }
+  });
+  return errors;
 };
 
 export function Form({ submitTitle, footer, formElements }: FormProps) {
   const classes = useStyles();
+  const dispatch = useDispatch();
 
   return (
     <FinalForm
-      onSubmit={() => {}}
-      render={({ handleSubmit, pristine, submitting }) => (
-        <form onSubmit={handleSubmit}>
-          <Grid container spacing={2}>
-            {formElements.map(({ label, inputProps }) => (
-              <Grid item xs={12} sm={12} key={label}>
-                <Field name={label}>
-                  {({ input }) => (
-                    <>
-                      <label className={classes.label}>{label}</label>
-                      <Input {...input} {...inputProps} />
-                    </>
-                  )}
-                </Field>
-              </Grid>
-            ))}
-            <Button
-              type="submit"
-              fullWidth
-              variant="contained"
-              disabled={submitting || pristine}
-              className={classes.submit}
-            >
-              {submitTitle}
-            </Button>
-            <Divider className={classes.divider} variant="fullWidth" />
-            <p className={classes.newCustomer}>{footer.title}</p>
-            <Button
-              fullWidth
-              href={footer.href}
-              variant="outlined"
-              className={classes.register}
-            >
-              {footer.label}
-            </Button>
-          </Grid>
-        </form>
-      )}
+      onSubmit={(payload) => {
+        submitTitle === "sign in"
+          ? dispatch(
+              fetchUserRequest({
+                login: payload.email,
+                password: payload.password,
+              })
+            )
+          : dispatch(
+              registerUserRequest({
+                mail: payload.email,
+                password: payload.password,
+                userType: "Customer",
+              })
+            );
+      }}
+      validate={validator(formElements)}
+      render={({ handleSubmit, pristine, submitting, hasValidationErrors }) => {
```

Comment:
Let's assign the function to a const

Reply:
Can we skip this one? It would be difficult to type this function, and It's a part of UI anyway